persistence.js

```diff
@@ -627,8 +657,94 @@ class RedisPersistence extends CachedPersistence {
     this._db.eval(luaScript, 4, [topic, groupTopic, clientTopic, SHAREDTOPICS], cb)
   }
 
+  fillSharedCache (cb) {
+    const that = this
+    that._db.smembers(SHAREDTOPICS, function (err, sharedTopics) {
+      if (err) {
+        return cb(err)
+      }
+      if (!sharedTopics) {
+        return cb(null)
+      }
+
+      that.sharedCachedQlobber = new Qlobber(qlobberOpts)
+      that.cachedSharedTopics = new Set(sharedTopics)
+      const pipeline = that._db.pipeline()
+      sharedTopics.forEach((topicFromResult) => {
+        that.sharedCachedQlobber.add(topicFromResult, topicFromResult)
+        pipeline.smembers(topicFromResult)
+      })
+
+      // Execute pipeline
+      const allClientTopics = []
+      pipeline.exec((err, replies) => {
+        if (err) {
+          cb(err)
+        } else {
+          // Map replies to topics
+          that.cachedSharedTopicsToGroups = new Map()
+          replies.forEach((reply, index) => {
+            const topic = sharedTopics[index]
+            that.cachedSharedTopicsToGroups.set(topic, new Set(reply[1]))
+            if (reply[1] && reply[1].length > 0) {
```

Comment:
Shouldn't you check it before you create set from it?
```suggestion
            if (reply[1] && reply[1].length > 0) {
                that.cachedSharedTopicsToGroups.set(topic, new Set(reply[1]))
```